src/dependencies/DependencyMap.js

```diff
@@ -135,12 +135,13 @@ export function calculateForDispatch(
 
 export function calculateForPropsChange(
   dependencies: DependencyMap,
-  props: Object
+  props: Object,
+  state: ?Object
 ): Object {
   return oFilterMap(
     dependencies,
     (dep) => dep.deref && dep.deref.length > 0,
-    (dep) => calculate(dep, props)
+    (dep) => calculate(dep, props, state)
```

Comment:
mmm so I think the only thing I would change is that, if the the deref doesn't declare a `state` argument (i.e. the argument at index `1`) it shouldn't receive it.

```javascript
return oFilterMap(
  ...,
  (dep) => {
    if (dep.deref.length === 1) {
      return caluclate(dep, props);
    }
    return calculate(dep, props, state);
  }
);
```